R/dataset.R

```diff
@@ -329,12 +329,39 @@ setMethod("allVariables<-", c("CrunchDataset", "VariableCatalog"),
 setMethod("hidden", "CrunchDataset", function (x) hidden(allVariables(x)))
 
 
-webURL <- function (x) {
+APIToWebURL <- function (x) {
     ## URL to view this dataset in the web app
     stopifnot(is.dataset(x))
     return(paste0(absoluteURL("/", getOption("crunch.api")), "dataset/", id(x)))
 }
 
+webToAPIURL <- function (url) {
+    id <- sub("^https.*?/dataset/([0-9a-f]+)/.*$", "\\1", url)
```

Comment:
It wouldn't as-is ("datasets" vs. "dataset", subtle difference in the whaam URL): https://github.com/Crunch-io/rcrunch/blob/master/R/variable.R#L169

Could make the regexp in that line match on either if you think that's cleaner.